webui/react/src/components/AddUsersToGroupsModal.tsx

```diff
@@ -0,0 +1,78 @@
+import Form from 'components/kit/Form';
+import { Modal } from 'components/kit/Modal';
+import Select, { Option } from 'components/kit/Select';
+import { makeToast } from 'components/kit/Toast';
+import { assignMultipleGroups } from 'services/api';
+import { V1GroupSearchResult } from 'services/api-ts-sdk';
+import handleError from 'utils/error';
+
+const GROUPS_NAME = 'groups';
+
+type FormInputs = {
+  [GROUPS_NAME]: number[];
+};
+
+interface Props {
+  userIds: number[];
+  groupOptions: V1GroupSearchResult[];
+  clearTableSelection: () => void;
+  fetchUsers: () => void;
+}
+
+const AddUsersToGroupsModalComponent = ({
+  userIds,
+  groupOptions,
+  clearTableSelection,
+  fetchUsers,
+}: Props): JSX.Element => {
+  const [form] = Form.useForm<FormInputs>();
+
+  const onSubmit = async () => {
+    const values = await form.validateFields();
+
+    try {
+      const groupIds = Array.from(
+        new Set(values[GROUPS_NAME].flatMap((v) => (v !== undefined ? [v] : []))),
+      );
+      await assignMultipleGroups({ addGroups: groupIds, removeGroups: [], userIds });
+      makeToast({
+        title: 'Successfully set roles',
```

Comment:
typo

Reply:
Should this be `Successfully added to groups`?

Reply:
ah gotcha. thanks